src/c_kzg_4844.c

```diff
@@ -989,22 +1006,28 @@ C_KZG_RET verify_kzg_proof(bool *out,
 }
 
 static C_KZG_RET evaluate_polynomial_in_evaluation_form(BLSFieldElement *out, const Polynomial p, const BLSFieldElement *x, const KZGSettings *s) {
-  fr_t tmp, *inverses_in, *inverses;
+  C_KZG_RET ret;
+  fr_t tmp;
+  fr_t *inverses_in = NULL;
+  fr_t *inverses = NULL;
   uint64_t i;
   const fr_t *roots_of_unity = s->fs->roots_of_unity;
 
-  TRY(new_fr_array(&inverses_in, FIELD_ELEMENTS_PER_BLOB));
-  TRY(new_fr_array(&inverses, FIELD_ELEMENTS_PER_BLOB));
+  ret = new_fr_array(&inverses_in, FIELD_ELEMENTS_PER_BLOB);
+  if (ret != C_KZG_OK) goto out_error;
+  ret = new_fr_array(&inverses, FIELD_ELEMENTS_PER_BLOB);
+  if (ret != C_KZG_OK) goto out_error;
+
   for (i = 0; i < FIELD_ELEMENTS_PER_BLOB; i++) {
     if (fr_equal(x, &roots_of_unity[i])) {
       *out = p[i];
-      free(inverses_in);
-      free(inverses);
-      return C_KZG_OK;
+      goto out_error;
```

Comment:
this should set `ret` to `C_KZG_OK` first. the label is also a bit misleading here.

Reply:
Oh good catch. And yeah, I'll fix that.

Reply:
Maybe the labels should all be `out`? I think there's only one case where the `_error` part is needed to distinguish from `success`?

Reply:
Sounds good to me. Fixed. For some reason I thought there would be a conflict with the `out` parameter.